### PR TITLE
Add new "sync-with-master" action

### PR DESCRIPTION
**Pull request by prakashsurya**

### CODE REVIEW

File: sync-with-master/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+#
+# We need the full git repository history in order to do the git-merge.
+#
+git fetch --unshallow
+
+#
+# In order to open a pull request, we need to push to a remote branch.
+# To avoid conflicting with existing remote branches, we use branches
+# within the "sync-with-master" namespace.
```
Comment (shartse) [inline sync-with-master/entrypoint.sh:23]: Looks like a mismatch between the name of the branch in the comment vs the code. I'm fine either way, but let's make sure they match.

**Review by shartse — APPROVED** [review]

File: sync-with-master/entrypoint.sh (hunk 1 above)
Comment (prakashsurya) [reply sync-with-master/entrypoint.sh:23]: Good catch; let me fix that.

**Review by sebroy — APPROVED** [review]

**Comment by pzakha** [comment]
How is this repository going to be used?

**Comment by prakashsurya** [comment]
@pzakha see here: https://github.com/delphix/sdb/pull/217

**Comment by pzakha** [comment]
Ah, that's really neat. I'm wondering if we can do something similar for adding checkstyle to our various repositories (i.e. have some action repo that runs shfmt, shellcheck, ansible check, flake8, etc.)

**Comment by prakashsurya** [comment]
>  I'm wondering if we can do something similar for adding checkstyle to our various repositories

We definitely can, we just haven't done that yet.

We already use github actions to do this (e.g. see [here](https://github.com/delphix/sdb/blob/master/.github/workflows/main.yml) and [here](https://github.com/delphix/delphix-platform/blob/master/.github/workflows/main.yml)), but we haven't yet pulled them into this repository such that they can be reused in multiple repositories (so we've duplicated some of that logic across multiple repositories, so far).

**Pull request merged**